### PR TITLE
Wip/debug_2

### PR DESCRIPTION
**Pull request by Mehrad0711**
Changes:
- Bootleg pipeline and Dockefile updates accompanying https://github.com/stanford-oval/genienlp/pull/83  (full pipeline is yet to come)
- Remaining improvements for translation and paraphrasing in SPL pipelines

### CODE REVIEW

**Comment by gcampax** [comment]
45 commits??? You can't be serious...

**Comment by Mehrad0711** [comment]
:)) You expected more?
Most are for SPL changes. I can close this and make a separate PR for Bootleg and SPL changes.

**Comment by gcampax** [comment]
Let's get this reviewed, but next time you gotta split into multiple smaller PRs. Especially if this is not work happening for a paper!

File: components/evaluate-spl.yaml [hunk 1]
```diff
@@ -47,52 +49,79 @@ implementation:
...
+        aws s3 sync --no-progress ${s3_database_dir} ./database/ --exclude '*'  --include '*type2id.json' --include '*pretrained_bert_models*' --include '*entity_mappings/config.json' --include '*entity_mappings/qid2title.json' --include '*entity_mappings/qid2eid.json' --include '*canonical2type.json' --include '*es_*.json' --include '*wikidatatitle_to_typeid_0905.json' --include '*wikidataqid_to_bootlegtypeid.json' --include '*entityQID_to_wikidataTypeQID.json' --include '*alias2qids_wiki*' --include "*${bootleg_model}/*.json" --include "*${bootleg_model}/*.pt"
+      fi
+
       IFS='+'; read -ra PLS <<<"$pred_languages"; IFS=' '
+      IFS='+'; read -ra ES <<<"$eval_set"; IFS=' '
+
       echo "${PLS[@]}"
 
       export TZ=America/Los_Angeles
+      export GENIENLP_DATABASE_DIR=./database/
```
Comment (gcampax) [inline components/evaluate-spl.yaml:62]: You're going to make this change to the regular evaluate job too right?

File: pipelines/bootleg.py [hunk 2]
```diff
@@ -82,15 +79,14 @@ def bootleg(
...
      )
     (add_env(add_ssh_volume(bootleg_op), bootleg_env)
      .add_toleration(V1Toleration(key='nvidia.com/gpu', operator='Exists', effect='NoSchedule'))
      .add_node_selector_constraint('beta.kubernetes.io/instance-type', f'p3.{2 * bootleg_num_gpus}xlarge')
-     .add_volume(V1Volume(name='tensorboard',
-                          persistent_volume_claim=V1PersistentVolumeClaimVolumeSource('tensorboard-research-kf')))
     .add_volume(V1Volume(name='shm', empty_dir=V1EmptyDirVolumeSource(medium='Memory')))
      )
+
+    bootleg_op.container.set_image_pull_policy('Always')
```
Comment (gcampax) [inline pipelines/bootleg.py]: Do we need this? We don't, right?

File: sync-repos.sh [hunk 3]
```diff
@@ -2,24 +2,27 @@ set -ex
...
 fi
 
+
+cd /opt/genienlp/
+GENIENLP_HEAD=`git rev-parse HEAD`
+if [ -n "${GENIENLP_VERSION}" ] && [ "${GENIENLP_VERSION}" != "${GENIENLP_HEAD}" ]; then
+  git fetch
+  git checkout ${GENIENLP_VERSION}
+  pip3 install -e .
+fi
```
Comment (gcampax) [inline sync-repos.sh]: Is this change in order right? Pip installing genienlp will uninstall bootleg, and override any bootleg version you specify.

**Review by gcampax — COMMENTED** [review]
Couple questions. I haven't reviewed the SPL bits, those are yours and I assume those are tested.

File: components/evaluate-spl.yaml (hunk 1 above)
Comment (Mehrad0711) [reply components/evaluate-spl.yaml:62]: Yeah. Once we figure out the bootleg part, I will make an end2end pipeline with no spl in the middle!

File: pipelines/bootleg.py (hunk 2 above)
Comment (Mehrad0711) [reply pipelines/bootleg.py]: Not really but it's safer to have it. Pulling is fast compared to the job run time.

File: sync-repos.sh (hunk 3 above)
Comment (Mehrad0711) [reply sync-repos.sh]: Before the order mattered cause bootleg pinned all package versions. I relaxed this so genienlp and bootleg agree on shared packages. I can still change the order to match the dockerfile.

Comment (Mehrad0711) [reply sync-repos.sh]: In fact, let me do a test.

File: pipelines/bootleg.py (hunk 2 above)
Comment (gcampax) [reply pipelines/bootleg.py]: Yes but there are other places where we don't specify it. This makes it sloppy, which is the opposite of safe. Better to fail early and often, than to fail later!

Comment (Mehrad0711) [reply pipelines/bootleg.py]: We should add it to other pipelines too then (it's already in all SPL pipelines). I really prefer to keep it but can remove if you insist..

Comment (gcampax) [reply pipelines/bootleg.py]: Yes, please remove it. It's recommended by Kubernetes authors not to use it.

**Pull request merged**